### PR TITLE
Define LSRA's RefType enum via macros

### PR DESCRIPTION
**Pull request by mikedn**

### CODE REVIEW

**Comment by BruceForstall** [comment]
@sivarv @CarolEidt PTAL
/cc @dotnet/jit-contrib

File: src/jit/lsra.h [hunk 1]
```diff
@@ -77,20 +77,9 @@ struct LsraBlockInfo
...
-    RefTypeFixedReg            = 0x10,
-    RefTypeExpUse              = (0x20 | RefTypeUse),
-    RefTypeParamDef            = (0x10 | RefTypeDef),
-    RefTypeDummyDef            = (0x20 | RefTypeDef),
-    RefTypeZeroInit            = (0x30 | RefTypeDef),
-    RefTypeUpperVectorSaveDef  = (0x40 | RefTypeDef),
-    RefTypeUpperVectorSaveUse  = (0x40 | RefTypeUse),
-    RefTypeKillGCRefs          = 0x80,
-    RefTypeBound,
+#define DEF_REFTYPE(e, v, s) e = v,
```
Comment (pgavlin) [inline src/jit/lsra.h]: Nit: more expressive names in this macro would be nice (e.g. `DEF_REFTYPE(memberName, memberValue, unused) memberName = memberValue)`

Comment (pgavlin) [inline src/jit/lsra.h]: (same goes for the other cases)

**Comment by pgavlin** [comment]
LGTM aside from one nit.

**Comment by CarolEidt** [comment]
Very nice!

**Comment by mikedn** [comment]
Update macro parameter names at @pgavlin suggestion

**Pull request merged**